### PR TITLE
Remove references to schema1 pacakge from handlers

### PR DESCRIPTION
**Pull request by milosgajdos**
This is a rebase of https://github.com/distribution/distribution/pull/3984 taken over from @DavidSpek with his permission due to GH permissions sadness (https://github.com/distribution/distribution/pull/3984#issuecomment-1681327720).

### CODE REVIEW

File: registry/handlers/api_test.go [hunk 1]
```diff
@@ -421,7 +420,8 @@ func TestCatalogAPI(t *testing.T) {
 
 	envWithLessImages := newTestEnv(t, false)
 	for _, image := range allCatalog[0:(maxEntries - 1)] {
-		createRepository(envWithLessImages, t, image, "sometag")
+		imageName, _ := reference.WithName(image)
```
[the remark this answers is not in the document]
Comment (milosgajdos) [reply registry/handlers/api_test.go]: Done

File: registry/handlers/manifests.go [hunk 2]
```diff
@@ -174,56 +165,6 @@ func (imh *manifestHandler) GetManifest(w http.ResponseWriter, r *http.Request)
...
-
-		manifest, err = manifests.Get(imh, manifestDigest)
-		if err != nil {
-			if _, ok := err.(distribution.ErrManifestUnknownRevision); ok {
-				imh.Errors = append(imh.Errors, v2.ErrorCodeManifestUnknown.WithDetail(err))
-			} else {
-				imh.Errors = append(imh.Errors, errcode.ErrorCodeUnknown.WithDetail(err))
-			}
-			return
-		}
```
Comment (deleteriousEffect) [inline registry/handlers/manifests.go:215]: Removing selecting a single image from a manifest list when that is not supported is out of the scope of simply dropping support for schema1. If we want to remove this, we should do it in its own MR.

**Review by deleteriousEffect — CHANGES_REQUESTED** [review]

File: registry/handlers/manifests.go (hunk 2 above)
Comment (milosgajdos) [reply registry/handlers/manifests.go:215]: Arrgh, excellent spot. I was focusing on the broken tests when I took this over and this skipped past my attention. Let me fix

Comment (milosgajdos) [reply registry/handlers/manifests.go]: Ok this should be fixed, now. PTAL @deleteriousEffect

**Review by deleteriousEffect — APPROVED** [review]
